File: 1248-count-number-of-nice-subarrays/main.py
class Solution:
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:

        
        count = 0
        odd_count = 0
        hash_map = {0: 1}

        for num in nums:
            if num & 1 != 0:
                odd_count += 1
                
            if odd_count - k in hash_map:
                count += hash_map[odd_count - k]

            if odd_count in hash_map:
                hash_map[odd_count] += 1
            else:
                hash_map[odd_count] = 1

        return count

File: 1248-count-number-of-nice-subarrays/main.rs
use std::collections::HashMap;

impl Solution {
    pub fn number_of_subarrays(nums: Vec<i32>, k: i32) -> i32 {
        
        let mut map: HashMap<i32, i32> = HashMap::new();
        map.insert(0, 1);

        let (mut count, mut odd_count) = (0, 0);

        for &num in nums.iter() {
            
            if num & 1 != 0 {
                odd_count += 1;
            }
            
            if let Some(value) = map.get(&(odd_count - k)) {
                count += *value;
            }

            *map.entry(odd_count).or_insert(0) += 1;
        }

        count
    }
}
